# CNAC
Cooccurnce Network Analysis for Compositional data

CNAC is a package for the generation and analysis of cooccurence networks with compositional data. Data generated by many next gen sequencing experiments is compositional (is a subsampling of the total community) which violates assumptions of typical cooccurence network analysis techniques. 16S sequencing data is often very compositional in nature so methods such as SparCC (https://bitbucket.org/yonatanf/sparcc) have been developed for studying correlations microbiome data. CNAC is designed with compositional data in mind and so provides multiple correlation measures including SparCC.

This tool has two parts called 'within' and 'between'. The 'within' method takes as input BIOM formatted files (http://biom-format.org/) and forms cooccurence networks using a user specified correlation metric and edge cutoff. From this network CNAC also finds modules of cooccuring obeservations in the cooccurence network by finding k-clique communities as implemented in networkx (https://networkx.github.io/). Modules are summarized and a new biom table with observations contained in modules collapsed into single observations are returned. This biom table along with a list of modules and their contents are output.  A gml file of the network that can be opened using network visualization tools such as cytoscape (http://www.cytoscape.org/) is created which contains all observation metadata provided in the input biom file as well as module information.

The 'between' method takes two biom tables as input and calculates all pairwise correlations between the tables using a selection of correlation metrics. A gml correlation network is output as well as a file containing statistics and p-values of all correlations.

## Installation
Download the repository and go to the top directory in it and simply enter:
```
python setup.py install
```
NOTE: This installs the latest version of pysurvey on pip. This is not the most current version of as of 2/22/2016. To get this version you need to go to https://bitbucket.org/yonatanf/pysurvey as install pysurvey using there directions. We are working on a work around for this.

## Example usage:

'within' mode:
```
correl_networks.py within -i example_table.biom -o output_folder/ -m sparcc --min_p .05
```

'between' mode:
```
correl_networks.py between -1 example_table1.biom -2 example_table2.biom -o output_folder/ -m spearman --min_p .05
```

